lib/function/algebra/simplify/simplifyCore.js

```diff
@@ -42,13 +42,29 @@ function factory(type, config, load, typed, math) {
    *     The expression to be simplified
    */
   function simplifyCore(node) {
-    if (type.isOperatorNode(node) && node.args.length <= 2) {
+    if (type.isOperatorNode(node) && node.isUnary()) {
       var a0 = simplifyCore(node.args[0]);
-      var a1 = node.args[1] && simplifyCore(node.args[1]);
+
       if (node.op === "+") {
-          if (node.args.length === 1) {
-            return node.args[0];
+        return a0;
+      }
+
+      if (node.op === '-' && node.fn === "unaryMinus") {
```

Comment:
Should `node.fn === "unaryMinus"` be replaced by `node.isUnary()`. SimplyCore should be agnostic to `node.fn`.

Same with other checks of `node.fn` in simplifyCore

Reply:
Thanks, that's a good point. the old `node.fn == 'unaryMinus` is redundant when using `node.isUnary()`